gnome-program.c: In function 'gnome_program_init_paramv':
gnome-program.c:2002: error: invalid initializer
---
 libgnome/gnome-program.c |   19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

Index: libgnome-2.32.2+20170910.8487c2b4/libgnome/gnome-program.c
===================================================================
--- libgnome-2.32.2+20170910.8487c2b4.orig/libgnome/gnome-program.c
+++ libgnome-2.32.2+20170910.8487c2b4/libgnome/gnome-program.c
@@ -2130,6 +2130,19 @@ gnome_program_initv (GType type,
  *
  * Since: 2.8
  */
+static GnomeProgram *Matthew_Wilcox_fixed_libgnome_without_touching_GNOME_source_code(GType type,
+		const char *app_id, const char *app_version, const GnomeModuleInfo *module_info,
+		int argc, char **argv, guint nparams, GParameter *params, ...)
+{
+	va_list empty_va;
+	GnomeProgram *ret;
+
+	va_start(empty_va, params);
+	ret = gnome_program_init_common(type, app_id, app_version, module_info,
+				      argc, argv, NULL, empty_va, nparams, params);
+	va_end(empty_va);
+	return ret;
+}
 GnomeProgram*
 gnome_program_init_paramv (GType type,
                            const char *app_id, const char *app_version,
@@ -2137,9 +2150,5 @@ gnome_program_init_paramv (GType type,
                            int argc, char **argv,
                            guint nparams, GParameter *params)
 {
-    va_list empty_va = NULL;
-
-    return gnome_program_init_common (type, app_id, app_version, module_info,
-				      argc, argv, NULL, empty_va, nparams, params);
-
+    return Matthew_Wilcox_fixed_libgnome_without_touching_GNOME_source_code(type, app_id, app_version, module_info, argc, argv, nparams, params);
 }
